WordPress/Classes/Services/MediaImageService.swift

```diff
@@ -0,0 +1,196 @@
+import Foundation
+
+/// A service for handling the process of retrieving and generating thumbnail images
+/// for existing Media objects, whether remote or locally available.
+final class MediaImageService: NSObject {
```

Comment:
This class is a replacement for [MediaThumbnailService.swift](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/Services/MediaThumbnailService.swift), which I haven't touched to make sure it doesn't introduce any regressions. 

The previous implementation was pretty hard to follow because of all of the closures, so it was a perfect use-case for async/await. But the main reason I added it was to support cancellation to address https://github.com/wordpress-mobile/WordPress-iOS/issues/14528.

It also takes advantage of `Task.detached` for some background tasks. The great thing about the new concurrency system is that it automatically limits the number of threads.